app/src/main/java/com/concordia/mcga/helperClasses/DatabaseHelper.java

```diff
@@ -29,50 +28,50 @@
      */
     private DatabaseHelper(Context context) {
         super(context, DATABASE_NAME, null, 1);
-        DatabaseHelper.context = context;
+        this.context = context;
     }
 
     /**
      * Initializes values for database creation
      * @param context Interface to global information about an application environment.
      */
-    public static void setupDatabase(Context context) {
-        if (db == null) {
-            DatabaseHelper.context = context;
+    public static boolean setupDatabase(Context context) {
+        if (databaseHelper == null) {
             databasePath = context.getFilesDir().getPath() + "/../databases/";
+            databaseHelper = new DatabaseHelper(context);
+            return true;
+        } else {
+            return false;
         }
     }
 
     /**
      * @return A DatabaseHelper object
      */
-    public static DatabaseHelper getInstance() {
-        if (context == null) {
-            return null;
-        }
-        if (db == null) {
-            return databaseHelper = new DatabaseHelper(context);
+    public static DatabaseHelper getInstance() throws MCGADatabaseException {
+        if (databaseHelper == null) {
+            throw new MCGADatabaseException("Please setup the Database first.");
         } else {
             return databaseHelper;
         }
 
     }
 
     /**
-     *
-     * @return The database
+     * @return - an instance of a readable Database <b>DO NOT FORGET to call {@link
+     * SQLiteDatabase#close()}</b> after you are done!
      */
-    public static SQLiteDatabase getDb() {
-        return db;
+    public SQLiteDatabase getDb() {
+        return databaseHelper.getReadableDatabase();
     }
 
     /**
      * Creates a empty database on the system and rewrites it with our own database.
      */
-    public void createDatabase() throws IOException {
+    public void createDatabase(boolean force) throws IOException {
```

Comment:
The only reason for this flag for debugging purposes.
When we are working on the application, it will be set to true. When demoing the application, we would set to false, to speed up the application's boot time. I found this to be the simplest way to toggle between dev/prod states.

The only possible improvement I can see is making the flag in the MainActivity a constant. What do you think?